src/main/scala/lms/core/traversal.scala

```diff
@@ -263,6 +263,8 @@ abstract class CPSTraverser extends Traverser {
 
 class CompactTraverser extends Traverser {
 
+  val mustInline = mutable.HashMap[Sym, Node]()
```

Comment:
That's scary

Reply:
This is for library Macros that serves as constant (such as MPI_COMM_WORLD). It is not wrong if they are not inlined, but more correct if they are.